cohortextractor/query_engines/mssql.py

```diff
@@ -171,12 +170,21 @@ def get_query_nodes_from_category_definitions(self, definitions, query_nodes=Non
         """
         Get all the referenced query nodes for category definitions.  If a category
         definition (Comparator) has a LHS which is not itself a Comparator,
-        it will be a query node
+        it will be a query node or a function-generated node
         """
         query_nodes = query_nodes or set()
         for definition in definitions:
             if isinstance(definition.lhs, Value):
-                query_nodes.add(definition.lhs)
+                # A ValueFromFunction is not itself a QueryNode, so we look for referenced
+                # nodes in its arguments
```

Comment:
I think this is just another signal that our type hierarchy is messed up, but looking at the code I see that a `ValueFromFunction`_is_ a `QueryNode` (it's a subclass of `Value`, which is a subclass of `QueryNode`).

Reply:
You're right, it is a QueryNode, but it's not a of a thing-we-need-to-add-temp-tables-for.  I'll update the comment